### PR TITLE
Feat/conditional writes

### PR DESCRIPTION
**Pull request by byo**

### CODE REVIEW

File: src/master/develop/reading.md [hunk 1]
```diff
@@ -367,14 +382,95 @@ const IMMUDB_PWD = 'immudb'
...
+It can be then used to ensure consistent state of data inside the database.
+
+Following preconditions are supported:
+
+* MustExist - precondition checks if given key exists in the database
+* MustNotExist - precondition checks if given key does not exist in the database
+* NotModifiedAfterTX - precondition checks if given key was not modified after given transaction id
+
+*Note: Checks do not consider logical deletion of entries.
+E.g. if an entry was logically deleted, it will still pass the `MustExist` precondition check.*
```
Comment (jeroiraz) [inline src/master/develop/reading.md:420]: constraint validation will filter out deleted entries

File: src/master/develop/reading.md [hunk 2]
```diff
@@ -367,14 +382,95 @@ const IMMUDB_PWD = 'immudb'
...
+
+Entries used in precondition do not have to overlap with keys to be written.
+This allows one to create much more complex constraints such as ability to write only one key from a group of keys.
```
Comment (jeroiraz) [inline src/master/develop/reading.md:423]: not sure I follow the more advanced constraint sample

File: src/master/develop/reading.md [hunk 3]
```diff
@@ -367,14 +382,95 @@ const IMMUDB_PWD = 'immudb'
...
+Following preconditions are supported:
+
+* MustExist - precondition checks if given key exists in the database
+* MustNotExist - precondition checks if given key does not exist in the database
+* NotModifiedAfterTX - precondition checks if given key was not modified after given transaction id
+
+*Note: Checks do not consider logical deletion of entries.
+E.g. if an entry was logically deleted, it will still pass the `MustExist` precondition check.*
+
+Entries used in precondition do not have to overlap with keys to be written.
```
Comment (jeroiraz) [inline src/master/develop/reading.md:422]: what about the case of ensuring it's the first write to a key e.g. mustNotExist key1, set key1, value1?

File: src/master/develop/reading.md (hunk 2 above)
Comment (byo) [reply src/master/develop/reading.md:428]: I've split examples into few cases

**Review by jeroiraz — APPROVED** [review]

**Pull request merged**